blaze-client/src/main/scala/org/http4s/blaze/client/PoolManager.scala

```diff
@@ -111,24 +114,30 @@ private final class PoolManager[F[_], A <: Connection[F]](
   private def incrConnection(key: RequestKey): F[Unit] =
     F.delay {
       curTotal.getAndIncrement()
-      allocated.update(key, allocated.getOrElse(key, 0) + 1)
+      val update: BiFunction[RequestKey, Int, Int] = (_: RequestKey, v: Int) => v + 1
+      allocated.compute(key, update)
```

Comment:
I find this fascinating - the default behavior of `ConcurrentHashMap` on getting values on empty keys is about returning NPE. But if we try it in Scala with `Int` as a type of values, the '0' value will be returned then. So we don't need to check a possible NPE here.

Reply:
If one is interested in why this is so: https://github.com/scala/bug/issues/602#issuecomment-292349468